docs/content/concepts/terminology.md

```diff
@@ -1,100 +1,192 @@
 ---
 description: >
-  Contains the definitions shared across design documents around prototyping a kube-like control plane (in KCP).  This is
-  a derivative work of other design documents intended to frame terminology.  All future statements that may be changed by
-  designs is covered by those designs, and not duplicated here.
+  The Terminology document contains definitions used throughout the kcp project. We strongly recommend familiarizing
+  yourself with this terminology before proceeding. Note that some terms might get changed in the future, and that
+  will be covered by respective design documents.
 ---
 
 # Terminology
 
+This document contains definitions used throughout the kcp project. We strongly recommend familiarizing yourself with
+this terminology before proceeding.
+
+!!! note
+    Some terms might get changed as the project develops. This document will be kept up to date with all relevant
+    changes, while additional details and history of changes will be covered by respective design documents.
+
+## `kcp` server
+
+kcp server is a Kubernetes-like API server. It extends the Kubernetes API server to provide multi-tenancy and
+capabilities for building platforms on top of it. The kcp server can be subdivided into multiple "logical clusters"
+where different logical clusters can be used by different users in the organization.
+
+The kcp server only provides resources to accomplish this, such as `LogicalCluster`, `Workspace`, `APIBinding` and
+`APIExport`, on top of some core Kubernetes resources, like `ConfigMap` and `Secret`. It doesn't provide Kubernetes
+resources for managing and orchestrating workloads, such as Pods and Deployments. The list of resources provided by
+default can be found in the [Built-in APIs document](./apis/built-in).
+
+kcp follows the Kubernetes API semantics, and all users are required to adhere to these semantics. In other words,
+if you're building a platform on top of kcp, you'll be required to build Kubernetes-like resources and APIs, think of
+CustomResourceDefinitions (CRDs) and controllers. The difference is that in some cases you might have to use a
+kcp-specific resources instead of regular Kubernetes resources to accomplish something. For example, to define your
+own resource that can be used across different logical clusters, you'll have to use `APIResourceSchema` resource
+instead of `CustomResourceDefinition`.
```

Comment:
This reads a little strange. We don't force users.

```suggestion
kcp follows the Kubernetes API semantics in each logical cluster, i.e. kcp should be conformant to the subset of the Kubernetes conformance suite that applies to the APIs available in kcp. In other words, if you're building a platform on top of kcp, you'll be required to build Kubernetes-like resources and APIs, think of
CustomResourceDefinitions (CRDs) and controllers. The difference is that in some cases you might have to use a
kcp-specific resources instead of regular Kubernetes resources to accomplish something. For example, to define your
own resource that can be used across different logical clusters, you'll have to use `APIResourceSchema` resource
instead of `CustomResourceDefinition`.
```